tests/transports/test_all_plugins.py

```diff
@@ -1047,73 +916,50 @@ def test_put_get_empty_string_tree(custom_transport):
 
         # TODO : get doesn't retrieve empty files.
         # Is it what we want?
-        transport.gettree(remote_subfolder, retrieved_subfolder)
-
-        os.remove(os.path.join(local_subfolder, 'file.txt'))
-        os.rmdir(local_subfolder)
-        transport.remove(os.path.join(remote_subfolder, 'file.txt'))
-        transport.rmdir(remote_subfolder)
-        # If it couldn't end the copy, it leaves what he did on local file
-        # here I am mixing local with remote
-        assert 'file.txt' in transport.listdir('tmp3')
-        os.remove(os.path.join(retrieved_subfolder, 'file.txt'))
-        os.rmdir(retrieved_subfolder)
+        transport.gettree(str(remote_subfolder), str(retrieved_subfolder))
 
-        transport.chdir('..')
-        transport.rmdir(directory)
+        assert 'file.txt' in [p.name for p in retrieved_subfolder.iterdir()]
 
 
-def test_gettree_nested_directory(custom_transport):
+def test_gettree_nested_directory(custom_transport, remote_tmp_path: Path, tmp_path: Path):
     """Test `gettree` for a nested directory."""
-    with tempfile.TemporaryDirectory() as dir_remote, tempfile.TemporaryDirectory() as dir_local:
-        content = b'dummy\ncontent'
-        filepath = os.path.join(dir_remote, 'sub', 'path', 'filename.txt')
-        os.makedirs(os.path.dirname(filepath))
+    content = b'dummy\ncontent'
+    dir_path = remote_tmp_path / 'sub' / 'path'
+    dir_path.mkdir(parents=True)
+
+    file_path = str(dir_path / 'filename.txt')
+
+    with open(file_path, 'wb') as handle:
+        handle.write(content)
 
-        with open(filepath, 'wb') as handle:
-            handle.write(content)
+    with custom_transport as transport:
+        transport.gettree(str(remote_tmp_path / 'sub' / 'path'), str(tmp_path / 'sub' / 'path'))
```

Comment:
Here I think we are supposed to check the functionality of `gettree`. Only this would be serve the purpose of the test. I understand, it was incompetent even before your changes :)
```suggestion
        transport.gettree(str(remote_tmp_path), str(tmp_path))
```

Reply:
cool! I didn't realize that, the tests passed with your change!